choreolib/src/main/java/choreo/trajectory/DifferentialSample.java

```diff
@@ -109,18 +116,52 @@ public ChassisSpeeds getChassisSpeeds() {
   @Override
   public DifferentialSample interpolate(DifferentialSample endValue, double timestamp) {
     double scale = (timestamp - this.t) / (endValue.t - this.t);
-    var interp_pose = getPose().interpolate(endValue.getPose(), scale);
+
+    // Integrate the acceleration to get the rest of the state, since linearly
+    // interpolating the state gives an inaccurate result if the accelerations are changing between
+    // states
+    Matrix<N6, N1> initialState = VecBuilder.fill(x, y, heading, omega, vl, vr);
+
+    double width = (vr - vl) / omega;
+    BiFunction<Matrix<N6, N1>, Matrix<N2, N1>, Matrix<N6, N1>> f =
+        (state, input) -> {
+          //  state =  [x, y, θ, ω, vₗ, vᵣ]
+          //  input =  [aₗ, aᵣ]
+          //
+          //  v = (vₗ + vᵣ)/2
+          //  ω = (vᵣ − vₗ)/width
+          //  α = (aᵣ − aₗ)/width
+          //
+          //  ẋ = v cosθ
+          //  ẏ = v sinθ
+          //  θ̇ = ω
+          //  v̇ₗ = aₗ
+          //  v̇ᵣ = aᵣ
+          //  ω̇ = α
+          var heading = state.get(2, 0);
+          var vl = state.get(4, 0);
+          var vr = state.get(5, 0);
+          var ar = state.get(0, 1);
+          var al = state.get(0, 0);
+          var v = (vl + vr) / 2;
+          var ω = (vl - vr) / width;
+          var α = (ar - al) / width;
+          return VecBuilder.fill(v * Math.cos(heading), v * Math.sin(heading), ω, al, ar, α);
+        };
+
+    double τ = timestamp - this.t;
+    var sample = NumericalIntegration.rkdp(f, initialState, VecBuilder.fill(al, ar), τ);
 
     return new DifferentialSample(
         MathUtil.interpolate(this.t, endValue.t, scale),
-        interp_pose.getX(),
-        interp_pose.getY(),
-        interp_pose.getRotation().getRadians(),
-        MathUtil.interpolate(this.vl, endValue.vl, scale),
-        MathUtil.interpolate(this.vr, endValue.vr, scale),
-        MathUtil.interpolate(this.omega, endValue.omega, scale),
-        MathUtil.interpolate(this.al, endValue.al, scale),
-        MathUtil.interpolate(this.ar, endValue.ar, scale),
+        sample.get(0, 0),
+        sample.get(1, 0),
+        sample.get(2, 0),
+        sample.get(4, 0),
+        sample.get(5, 0),
+        sample.get(3, 0),
+        sample.get(4, 0) / τ,
+        sample.get(5, 0) / τ,
```

Comment:
You could probably do something similar to the swerve acceleration calculation here.